tests/by-util/test_ls.rs

```diff
@@ -1376,6 +1376,84 @@ fn test_ls_long_symlink_color() {
     }
 }
 
+/// This test is for "ls -l --color=auto|--color=always"
+/// We use "--color=always" as the colors are the same regardless of the color option being "auto" or "always"
+/// using 'ln' to create a dangling symlink
+/// Only tests whether the specific color of the target and the dangling_symlink are equal
+/// Modified version of the "test_ls_long_symlink_color" test. Please see that for more details
+#[test]
+fn test_ls_long_dangling_symlink_color() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir("dir1");
+    ts.ccmd("ln")
+        .arg("-s")
+        .arg("foo")
+        .arg("dir1/dangling_symlink")
+        .succeeds();
+
+    // The string between \x1b[ and m
+    type Color = String;
+
+    // The string between the color start and the color end is the file name itself.
+    type Name = String;
+
+    let result = ts
+        .ucmd()
+        .arg("-l")
+        .arg("--color=always")
+        .arg("dir1/dangling_symlink")
+        .succeeds();
+
+    let stdout = result.stdout_str();
+
+    // Helper function to isolate the sequence in the string from where the ANSI C escape code begins
```

Comment:
Why is parsing required here? Shouldn't it be possible to do something like `.stdout_contains("\x1b[0m\x1b[40;31;01mdir1/dangling_symlink\x1b[0m -> \x1b[40;31;01minvalid\x1b[0m\n")`?
We don't need nice test-failure messages if a color is slightly wrong. Parsing is yet another opportunity to fail, so if we can get away with not parsing anything at all, then I would prefer that.

Reply:
Yeah this makes sense. I was avoiding regex due to my inexperience with it. I will switch to matching against a regex .